### PR TITLE
Add support for the ElectriQ iQ-wifiMOODL RGBW LED controller

### PR DESCRIPTION
**Pull request by ianbyte**
## Description:

This PR adds support for the ElectriQ iQ-wifiMOODL RGBW LED controller:
https://www.electriq.co.uk/p/iq-wifimoodl/electriq-dimmable-colour-changing-led-smart-mood-lamp-alexa-and-google-home-compatible

This lamp uses the TYWE3S module to transmit via UART to an unmarked MCU which drives 8 MOSFETs.
The MCU does not transmit; it only receives commands from the TYWE3S.
The MCU appears to use a modified/undocumented version of the TuyaMCU protocol.

## Checklist:
  - [X] The pull request is done against the latest dev branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR.
  - [X] The code change is tested and works on core Tasmota_core_stage
  - [X] The code change pass travis tests. **Your PR cannot be merged unless tests pass**
  - [X] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

### CODE REVIEW

**Comment by ascillato** [comment]
Thanks for sharing. 

This should be disabled as default in order to not increase flash usage for all users. If any user need this, he/she can self compile.

File: tasmota/language/bg-BG.h [hunk 1]
```diff
@@ -663,6 +663,7 @@
 #define D_SENSOR_CC1101_GDO0   "CC1101 GDO0"
 #define D_SENSOR_CC1101_GDO2   "CC1101 GDO2"
 #define D_SENSOR_HRXL_RX       "HRXL Rx"
+#define D_SENSOR_ELECTRIQ_MOODL "ElectriQ MOODL Tx"
```
Comment (arendst) [inline tasmota/language/bg-BG.h]: Text too long. See remark in file:
```
// tasmota_template.h - keep them as short as possible to be able to fit them in GUI drop down box
```
Just drop ElectriQ and keep ``MOODL Tx`` in all language files

File: BUILDS.md [hunk 2]
```diff
@@ -53,6 +53,7 @@
 | USE_SM16716           | - | - | x | x | x | - | x |
 | USE_SM2135            | - | - | x | x | x | - | x |
 | USE_SONOFF_L1         | - | - | x | x | x | - | x |
+| USE_ELECTRIQ_MOODL    | - | - | x | x | x | - | x |
```
Comment (arendst) [inline BUILDS.md]: Disable all by default as it is a niche product

**Comment by arendst** [comment]
Pls fix the above issues before merge.

BTW I love the tiny code addition for a new feature.

**Comment by arendst** [comment]
Thx.

Considering the tine code footprint (0k3) I'll enable it by default anyway.

**Comment by blakadder** [comment]
@ianbyte got a template for that lamp?

**Comment by arendst** [comment]
To be honest any template will do as long as some GPIO is configured as MOODL Tx.

But for completeness this should do:
```
{"NAME":"ElectriQ MOODL","GPIO":[0,201,0,0,0,0,0,0,0,0,0,0,0],"FLAG":0,"BASE":18}
```

**Comment by blakadder** [comment]
i hoped the gpio doesn't randomly change on that lamp :D

**Pull request merged**